pkg/actions/lua_test.go

```diff
@@ -88,6 +110,197 @@ func TestLuaRun(t *testing.T) {
 	}
 }
 
+func TestLuaRun_NetHttpDisabled(t *testing.T) {
+	h, err := actions.NewLuaHook(
+		actions.ActionHook{
+			ID:          "myHook",
+			Type:        actions.HookTypeLua,
+			Description: "na",
+			Properties: map[string]interface{}{
+				"script": `local http = require("net/http")`,
+			},
+		},
+		&actions.Action{
+			Name:        "",
+			Description: "",
+			On:          nil,
+			Hooks:       nil,
+		},
+		actions.Config{Enabled: true},
+		nil)
+	if err != nil {
+		t.Errorf("unexpedcted error: %v", err)
+	}
+	out := &bytes.Buffer{}
+	ctx := context.Background()
+	ctx = auth.WithUser(ctx, &model.User{
+		CreatedAt: time.Time{},
+		Username:  "user1",
+	})
+	err = h.Run(ctx, graveler.HookRecord{
+		RunID:            "abc123",
+		EventType:        graveler.EventTypePreCreateBranch,
+		RepositoryID:     "example123",
+		StorageNamespace: "local://foo/bar",
+		SourceRef:        "abc123",
+		BranchID:         "my-branch",
+		Commit: graveler.Commit{
+			Version: 1,
+		},
+		CommitID: "123456789",
+		PreRunID: "3498032432",
+		TagID:    "",
+	}, out)
+	const expectedErr = "module 'net/http' not found"
+	if err == nil || !strings.Contains(err.Error(), expectedErr) {
+		t.Fatalf("Error=%v, expected: '%s'", err, expectedErr)
+	}
+}
+
+func TestLuaRun_NetHttp(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		codeVal := r.URL.Query().Get("code")
+		statusCode, _ := strconv.Atoi(codeVal)
+		if statusCode != 0 {
+			w.WriteHeader(statusCode)
+		}
+		_, _ = fmt.Fprint(w, "hello-"+r.Method)
+		body, _ := io.ReadAll(r.Body)
+		if len(body) > 0 {
+			_, _ = w.Write([]byte{' '})
+			_, _ = w.Write(body)
+		}
+	}))
+	defer ts.Close()
+
+	tests := []struct {
+		Name     string
+		Script   string
+		Expected string
+	}{
+		{
+			Name: "simple_get",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request("` + ts.URL + `")
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-GET 200 OK",
+		},
+		{
+			Name: "simple_post",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request("` + ts.URL + `", "name=value")
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-POST name=value 200 OK",
+		},
+		{
+			Name: "simple_get_404",
```

Comment:
Maybe try the 404 with different address like `http://this.does.not.exist.treeverse.io`?

Reply:
Didn't want to call an http server on the web to get different status code - this is why I used the test server to simulate it.